addon/services/key-manager.js

```diff
@@ -91,23 +102,15 @@ export default Service.extend({
   },
 
   handleEvent(event) {
-    const isServiceDestroyed = get(this, 'isDestroyed') || get(this, 'isDestroying');
-
-    if (isServiceDestroyed) {
-      return false;
+    if (get(this, 'isDestroyed') || get(this, 'isDestroying')) {
+      return;
     }
 
-    set(this, 'event', event);
-
-    debounce(this, this._handleEvent, 100);
-  },
-
-  _handleEvent() {
-    const event = get(this, 'event');
     const isKeydown = event.type === 'keydown';
     const isKeyup = event.type === 'keyup';
 
     if (isKeydown || isKeyup) {
+      event.stopPropagation();
```

Comment:
Do we need this?

Reply:
I think we do but only if there is a match. Since otherwise, the event handler could be fired twice if there were two macros registered - one on the body and the other on an element contained in the body. I moved it further down so that we only stop on match. .